stan/math/opencl/matrix_cl.hpp

```diff
@@ -341,6 +347,47 @@ class matrix_cl<T, enable_if_arithmetic<T>> {
     }
   }
 
+  /**
+   * Constructs a const matrix_cl that contains a copy of the Eigen matrix on
+   * the OpenCL device. If the matrix already has a cached copy on the device,
+   * the cache is used and no copying is done. Changing the resulting matrix_cl
+   * would change cache, so do not do it! If changes are needed a copy must be
+   * made.
+   *
+   * @tparam R row type of input matrix
+   * @tparam C column type of input matrix
+   * @param A the Eigen matrix
+   * @param partial_view which part of the matrix is used
+   */
+  template <int R, int C>
+  static matrix_cl<T> constant(const Eigen::Matrix<T, R, C>& A,
+                               matrix_cl_view partial_view
+                               = matrix_cl_view::Entire) {
+#ifdef STAN_OPENCL_CACHE
```

Comment:
Do we still need this `STAN_OPENCL_CACHE` flag if caching will only be used when explicitly stated in memory transfer bound GPU functions? If anything, we might use a `STAN_OPENCL_NOCACHE` optionally. But I dont think event that is needed.

Reply:
Option to turn caching off with STAN_OPENCL_NOCACHE might come in handy in some cases.